include/nighthawk/adaptive_load/metrics_plugin.h

```diff
@@ -20,9 +22,10 @@ class MetricsPlugin {
    *
    * @param metric_name The name of the metric to retrieve. Must be supported by the plugin.
    *
-   * @return double The metric value, or 0.0 if the metric was unsupported or unavailable.
+   * @return StatusOr<double> The metric value, or an error status if the metric was unsupported or
+   * unavailable.
    */
-  virtual double GetMetricByName(const std::string& metric_name) PURE;
+  virtual Envoy::StatusOr<double> GetMetricByName(absl::string_view metric_name) PURE;
```

Comment:
Curious - why is this Envoy::StatusOr rather than absl::StatusOr?

Reply:
I had the same question but I don't think absl::StatusOr is open yet. I think that the reason it's in envoy is because it hasn't been made public.

Reply:
I found it's actually possible to use absl::StatusOr.